dashboard/src/components/FunctionView/FunctionView.tsx

```diff
@@ -66,10 +71,21 @@ class FunctionView extends React.Component<IFunctionViewProps, IFunctionViewStat
     const msg = JSON.parse(e.data);
     const deployment: IResource = msg.object;
     this.setState({ deployment });
+    // refetch pod name when deployment is available, in case in changed
+    // TODO: move deployment status into redux store
```

Comment:
Explanation of what's happening here:
- when clicking Save on a function, the Deployment goes into an unhealthy state and then healthy again. There's a lag between these stage changes
- We only want to get the pod name when it's healthy, that's when we can fetch the logs without that API call failing
- So we flip the state and only fetch the pod name if the deployment has become healthy after being unhealthy

I've added this TODO, because we should really move this logic into redux and away from the component in the future. I think we need to move all webhook watching into redux.